parquet-variant/src/variant/metadata.rs

```diff
@@ -128,15 +128,20 @@ impl VariantMetadataHeader {
 ///
 /// [`Variant`]: crate::Variant
 /// [Variant Spec]: https://github.com/apache/parquet-format/blob/master/VariantEncoding.md#metadata-encoding
-#[derive(Debug, Clone, Copy, PartialEq)]
+#[derive(Debug, Clone, PartialEq)]
 pub struct VariantMetadata<'m> {
     bytes: &'m [u8],
     header: VariantMetadataHeader,
-    dictionary_size: usize,
-    first_value_byte: usize,
+    dictionary_size: u32,
+    first_value_byte: u32,
     validated: bool,
 }
 
+#[cfg(test)]
+const _: () = if std::mem::size_of::<VariantMetadata>() != 32 {
+    panic!("VariantMetadata changed size, which will impact VariantList and VariantObject");
```

Comment:
It might help to give more context here
```suggestion
    // to ensure fast performance, we want to avoid increasing the size of VariantMetadata 
    panic!("VariantMetadata changed size, which will impact VariantList and VariantObject.");
```